src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/GetAzureFunctionsOperation.cs

```diff
@@ -0,0 +1,71 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.SqlTools.ServiceLayer.AzureFunctions.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureFunctions
+{
+    /// <summary>
+    /// Class to represent getting the Azure Functions in a file
+    /// </summary>
+    class GetAzureFunctionsOperation
+    {
+        const string functionAttributeText = "FunctionName";
+
+        public GetAzureFunctionsParams Parameters { get; }
+
+        public GetAzureFunctionsOperation(GetAzureFunctionsParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Gets the names of all the azure functions in a file
+        /// </summary>
+        /// <returns>the result of trying to get the names of all the Azure functions in a file</returns>
+        public GetAzureFunctionsResult GetAzureFunctions()
+        {
+            try
+            {
+                string text = File.ReadAllText(Parameters.filePath);
```

Comment:
Is this something that you could contribute to their extension? They might find it useful - and having all that kind of code in one place would be handy I think.

Reply:
I can look into contributing this to their extension later and can remove this if that gets added. The current exposed [api](https://github.com/microsoft/vscode-azurefunctions/blob/main/src/vscode-azurefunctions.api.d.ts) is pretty limited and it would take more time to get it added there, so I'll stick with this approach for now to get this out sooner.